docs/running-calyx/fud2/scripts.md

```diff
@@ -110,6 +167,39 @@ The name for an import is always just the basename of the script file, without a
 
 ## API
 
+### High Level Rhai
+
+#### defop
+
+```
+defop <op name>(<input1>: <input1 state>, <input2>: <input2 state> ...) >> <target1>: <target1 state>, <target2>, <target2 state> ... {
+    <statements>
+}
+```
+Defines an op with `<op name>` which runs `<statements>` to generate target states from input states. 
+
+The name, inputs, and targets are called the signature of the op. `<statements>` is called the body of the op.
+
+#### shell
+
+```
+shell(<string>)
+```
+When called in the body of an op, that op will run `<string>` as a shell command to generate its targets. It is an error to call `shell` outside of the body of an op. Additionally, it is an error to call `shell` in the body of an op in which `shell_deps` was called prior.
```

Comment:
While this is a good description of how to think about `shell` at a high level, maybe we should add a second paragraph to demystify what happens at the Ninja level. Something like this:

> In the generated Ninja code, `shell` will create both a `rule` wrapping the shell command and a `build` command that invokes that rule. When a `defop` contains multiple `shell` commands, fud2 automatically generates Ninja dependencies among the `build` commands to ensure that they run in order.